### PR TITLE
feat: create single testimonial retrieval endpoint

### PR DESCRIPTION
**Pull request by Andrea-Ozuem**
Created a route to handle retrieving testimonial by id
​

## Description

- Created Testimonial service in `services` folder with fetch(Class, id) endpoint to retrieve obj if it exists
- Used the fetch method to return valid obj else JSONResponse with 404
- Ensured JSONResponse with 401 was raised incase of invalid user credentials
- Created tests for the following cases:
    - Successful testimonial retrieval
    - Invalid user credentials
    - Testimonial id that doesn't exist.
​

## Related Issue (Link to issue ticket)

(https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/68)
​

## Motivation and Context

Adds required endpoint for app
​

## How Has This Been Tested?

Successful testimonial retrieval
- mocked calls to db for retrieving testimonial
- mocked calls to db for getting a user
- made a test get request to endpoint with user cred
- assertions on status code and mocked testimonial fields

 Invalid user credentials
- made a test get request to endpoint with no user cred (missing header)
- assertions on status code

Testimonial id that doesn't exist
- mocked calls to db for retrieving testimonial
- mocked calls to db for getting a user
- made a test get request with a different id from mocked testimonial with user cred
- made assertions on status code and message fields
​

## Screenshots (if appropriate - Postman, etc):

​

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/testimonial.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@testimonial.get('/{testimonial_id}', status_code=status.HTTP_200_OK)
+def get_testimonial(testimonial_id, db: Session = Depends(get_db), current_user: User = Depends(user_service.get_current_user)):
+    '''Endpoint to get testimonial by id'''
+
+    testimonial = testimonial_service.fetch(db, testimonial_id)
+    if testimonial and testimonial_id == testimonial.id:
+        return success_response(
+            status_code=200,
+            message='Testimonial {} retrieved successfully'.format(testimonial_id),
```
Comment (zxenonx) [inline api/v1/routes/testimonial.py:24]: use f-strings instead of format()

File: api/v1/routes/testimonial.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                'content': testimonial.content,
+                'ratings': testimonial.ratings, 
+            }
+        )
+    return JSONResponse(
+        status_code=404,
+        content={
+            "success": False,
+            "status_code": 404,
+            "message": "Testimonial {} not found".format(testimonial_id)
```
Comment (zxenonx) [inline api/v1/routes/testimonial.py:40]: use f-strings instead of format()

File: api/v1/services/testimonial.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    def fetch_all(self, db: Session):
+        '''Fetch all testimonial'''
+        pass
+
+    
+    def fetch(self, db: Session, id: str):
+        '''Fetches a single testimonial id'''
+
+        testimonial = check_model_existence(db, Testimonial, id)
+        return testimonial
```
Comment (zxenonx) [inline api/v1/services/testimonial.py:26]:
```suggestion
        return check_model_existence(db, Testimonial, id)
```

**Review by zxenonx — CHANGES_REQUESTED** [review]
Thanks @Andrea-Ozuem 
I left some comments

**Comment by Andrea-Ozuem** [comment]
> Thanks @Andrea-Ozuem I left some comments

Resolved!

**Review by zxenonx — APPROVED** [review]
Thanks @Andrea-Ozuem

**Pull request merged**